api/app.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from flask import Flask, jsonify, request, render_template, abort
+from data import events, ident_hash_list, pick_one
+
+def create_app():
+    settings = {
+        'DEBUG': True,
+    }
+    app = Flask(__name__)
+    app.config.update(settings)
+    return app
+
+app = create_app()
+
+def _get_events():
+    return events
+
+def _save_event(event):
+    events.append(event)
+
+@app.route('/events', methods=['GET'])
+def list_events():
+    events = _get_events()
+    return jsonify(dict(events=events))
+
+@app.route('/events/<int:id>', methods=['GET'])
+def get_event(id):
+    events = _get_events()
+    try:
+        event = next(event for event in events if event['id'] == id)
+        return jsonify(dict(event=event))
+    except:
+        abort(404)
+
+@app.route('/events', methods=['POST'])
+def add_event():
+    event = request.get_json()
+    events_len = len(_get_events())
+    event.update(dict(
+        id = events_len + 1,
+        created_at = datetime.utcnow().isoformat(),
+        ident_hash = ident_hash_list[pick_one]
```

Comment:
This should take the ident_hash from the POST request. A developer may want to target a specific book by the uuid.

Reply:
*update: make ident_hash a mandatory param